src/lib/tabs/tabs.ts

```diff
@@ -248,15 +249,123 @@ export class MdTabGroup {
   focusPreviousTab(): void {
     this.moveFocus(-1);
   }
+
+  private _createChangeEvent(index: number): MdTabChangeEvent {
+    const event = new MdTabChangeEvent;
+    event.index = index;
+    if (this._tabs && this._tabs.length) {
+      event.tab = this._tabs.toArray()[index];
+    }
+    return event;
+  }
+
+  /** Returns a unique id for each tab label element */
+  _getTabLabelId(i: number): string {
+    return `md-tab-label-${this._groupId}-${i}`;
+  }
+
+  /** Returns a unique id for each tab content element */
+  _getTabContentId(i: number): string {
+    return `md-tab-content-${this._groupId}-${i}`;
+  }
+
+  /**
+   * Sets the height of the body wrapper to the height of the activating tab if dynamic
+   * height property is true.
+   */
+  _setTabBodyWrapperHeight(e: number) {
+    if (!this._dynamicHeight) { return; }
+
+    this._renderer.setElementStyle(this._tabBodyWrapper.nativeElement, 'height',
+        this._tabBodyWrapperHeight + 'px');
+
+    // This conditional forces the browser to paint the height so that
+    // the animation to the new height can have an origin.
+    if (this._tabBodyWrapper.nativeElement.offsetHeight) {
+      this._renderer.setElementStyle(this._tabBodyWrapper.nativeElement, 'height', e + 'px');
+    }
+  }
+
+  /** Removes the height of the tab body wrapper. */
+  _removeTabBodyWrapperHeight() {
+    this._renderer.setElementStyle(this._tabBodyWrapper.nativeElement, 'height', '');
+  }
 }
 
+export type MdTabBodyActiveState = 'left' | 'center' | 'right';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-tab-body',
+  templateUrl: 'tab-body.html',
+  animations: [
+    trigger('position', [
+      state('left', style({transform: 'translateX(-100%)'})),
+      state('center', style({transform: 'translateX(0%)'})),
+      state('right', style({transform: 'translateX(100%)'})),
+      transition('* => *', animate('0.5s cubic-bezier(0.35, 0, 0.25, 1)')),
+    ])
+  ]
+})
+export class MdTabBody implements OnInit {
```

Comment:
Primarily code organization - portal attach/detach + animations would have added quite a bit to tab group and I don't think it would have been as readable about what is happening between all the interactions